turbopack/crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -2568,10 +2568,9 @@ async fn handle_free_var_reference(
                         ),
                         Default::default(),
                         match state.tree_shaking_mode {
-                            Some(TreeShakingMode::ModuleFragments)
-                            | Some(TreeShakingMode::ReexportsOnly) => {
-                                export.clone().map(ModulePart::export)
-                            }
+                            Some(
+                                TreeShakingMode::ModuleFragments | TreeShakingMode::ReexportsOnly,
+                            ) => export.clone().map(ModulePart::export),
```

Comment:
You also need the ModulePart of used_exports, but that works well once you have `TreeShakingMode::UnusedExportsRemoval`